gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationAlertServiceImpl.java

```diff
@@ -490,4 +477,22 @@ private UpdateAlertTriggerEntity convert(AlertTriggerEntity trigger) {
 
         return updating;
     }
+
+    private List<Filter> combineFilters(String applicationId, List<Filter> filtersToAdd) {
+        List<Filter> filters = new ArrayList<>();
+        filters.add(StringCondition.equals("application", applicationId).build());
+        if (filtersToAdd != null) {
+            filters.addAll(filtersToAdd);
+        }
+
+        return filters;
+    }
+
+    private List<Notification> combineNotifications(List<Notification>... notificationsToAdd) {
+        return Arrays.stream(notificationsToAdd).filter(Objects::nonNull).flatMap(Collection::stream).collect(Collectors.toList());
+    }
```

Comment:
I'm thinking about the pros/cons of adding a `addFilters` method in the `Alert` class, same for `addNotifications`. What's your opinion?

Reply:
As discussed together, we keep it internal